### ISSUE TITLE
VoiceOver doesn't read MatSelect option inside of a dialog

### ISSUE THREAD

**Issue by jelbourn** [angular/components#20694]
StackBlitz: https://stackblitz.com/edit/angular-q2ppnn?file=src%2Fapp%2Fdialog-overview-example-dialog.html

While `MatSelect` seems to work fine under normal circumstances, I can't get it to read the options when the select is inside of a dialog. I'm wondering if this might be a bug in VoiceOver, since it's surprising that the interaction would present different based on the dialog context.

**Comment by zelliott** [comment]
It's due to the `aria-modal="true"` on the dialog container.

It's not immediately obvious to me what the correct fix is. One idea is to construct the select's overlay within the aria-modal dialog container, but I imagine that would require non-trivial changes to overlays. Another idea is to try to use `aria-owns` to re-parent the select's overlay to the select itself.

**Comment by rameshworsht** [comment]
@wagnermaciel Could you please re-open this issue. https://github.com/angular/components/pull/20695 Does not fix this issue on Safari and Firefox.

**Comment by jelbourn** [comment]
Tracking via #20900 as a separate, distinct issue for Safari and Firefox and macOS.
